static/src/js/components/TemplateList/TemplateList.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Col, Row } from 'react-bootstrap'
+import pluralize from 'pluralize'
+import Page from '../Page/Page'
+import useAppContext from '../../hooks/useAppContext'
+import getTemplates from '../../utils/getTemplates'
+import Table from '../Table/Table'
+import EllipsisLink from '../EllipsisLink/EllipsisLink'
+import ErrorBanner from '../ErrorBanner/ErrorBanner'
+
+const TemplateList = ({ templateType }) => {
+  const { user } = useAppContext()
+  const { token } = user
+
+  const [templateList, setTemplateList] = useState([])
+  const [errors, setErrors] = useState()
+  const [loading, setLoading] = useState(true)
+
+  const { providerId } = user
+
+  useEffect(() => {
+    const fetchTemplates = async () => {
+      const { response, error } = await getTemplates(providerId, token)
+
+      setErrors(error)
+      setTemplateList(response)
+      setLoading(false)
+    }
+
+    fetchTemplates()
+  }, [])
+
+  const buildEllipsisLinkCell = useCallback((cellData, rowData) => {
+    const { id } = rowData
+
+    return (
+      <EllipsisLink to={`/templates/${pluralize(templateType).toLowerCase()}/${id}`}>
+        {cellData}
+      </EllipsisLink>
+    )
+  }, [])
+
+  const buildActionsCell = useCallback((cellData, rowData) => {
```

Comment:
Same comment regarding the empty dependency array. Just be sure this is doing what youre expecting, we don't want methods being memoized in ways we don't expect.